pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -431,26 +439,13 @@ func buildAutoMtlsSettings(
 		if len(subjectAltNamesToUse) == 0 {
 			subjectAltNamesToUse = serviceAccounts
 		}
-		// For backward compatibility, use metadata certs if provided
-		if proxy.Metadata.TLSClientRootCert != "" {
-			return buildMutualTLS(subjectAltNamesToUse, sniToUse, proxy), autoDetected
-		}
-
-		if tls.Mode != networking.ClientTLSSettings_ISTIO_MUTUAL {
-			return tls, userSupplied
-		}
 		return buildIstioMutualTLS(subjectAltNamesToUse, sniToUse), userSupplied
 	}
 
 	if meshExternal || !autoMTLSEnabled || serviceMTLSMode == model.MTLSUnknown || serviceMTLSMode == model.MTLSDisable {
 		return nil, userSupplied
 	}
 
-	// For backward compatibility, use metadata certs if provided
```

Comment:
We do not need to build mutual TLS with auto detected SAN and SNI because it does not work - When using custom certs, SAs are not used at all.